src/main/java/com/labsynch/labseer/service/SaltStructureServiceImpl.java

```diff
@@ -65,27 +65,106 @@ public Salt saveStructure(Salt salt) throws CmpdRegMolFormatException {
 
 	@Override
 	public Salt update(Salt salt) throws CmpdRegMolFormatException {
-		CmpdRegMolecule mol = chemStructureService.toMolecule(salt.getMolStructure());
-		salt.setOriginalStructure(salt.getMolStructure());
-		salt.setMolStructure(mol.getMolStructure());
-		salt.setFormula(mol.getFormula());
-		salt.setMolWeight(mol.getMass());
-		salt.setCharge(mol.getTotalCharge());
+		try
+		{
+			CmpdRegMolecule mol = chemStructureService.toMolecule(salt.getMolStructure());
+			salt.setOriginalStructure(salt.getMolStructure());
+			salt.setMolStructure(mol.getMolStructure());
+			salt.setFormula(mol.getFormula());
+			salt.setMolWeight(mol.getMass());
+			salt.setCharge(mol.getTotalCharge());
 
-		logger.debug("salt code: " + salt.getAbbrev());
-		logger.debug("salt name: " + salt.getName());
-		logger.debug("salt structure: " + salt.getMolStructure());
+			logger.debug("salt code: " + salt.getAbbrev());
+			logger.debug("salt name: " + salt.getName());
+			logger.debug("salt structure: " + salt.getMolStructure());
 
-		boolean updated = chemStructureService.updateStructure(mol, StructureType.SALT, salt.getCdId());
-		salt.merge();
+			boolean updated = chemStructureService.updateStructure(mol, StructureType.SALT, salt.getCdId());
+			salt.merge();
 
-		if (updated) {
-			return Salt.findSalt(salt.getId());
-		} else {
-			return null;
+			if (updated) {
+				return Salt.findSalt(salt.getId());
+			} else {
+				return null;
+			}
 		}
+		catch (Exception e)
```

Comment:
Might want to make this catch only CmpdRegMolFormatException

Reply:
I agree